### PR TITLE
Crowdfund : Add Buyer information / Additional information(forms) like POS

### PR DESCRIPTION
**Pull request by Nisaba**
Add the ability tho get buyer information or additional information via forms to a crowdfunf.
Same thing than POS.

### CODE REVIEW

**Comment by dstrukt** [comment]
Thanks for the PR! Will take a look this weekend - tagging @dennisreimann for review as well.

File: BTCPayServer/wwwroot/crowdfund/app.js [hunk 1]
```diff
@@ -220,7 +220,11 @@ document.addEventListener("DOMContentLoaded",function (ev) {
             },
             contribute() {
                 if (!this.active || this.loading) return;
-                
+
+                if (this.srvModel.hasFormForExtraValues) {
+                    window.location.href = this.srvModel.formUrl;
```
Comment (dennisreimann) [inline BTCPayServer/wwwroot/crowdfund/app.js:225]: This could be condensed to just formURL, also in the view model.

Comment (Nisaba) [reply BTCPayServer/wwwroot/crowdfund/app.js:225]: HasFormExtraValues boolean is set in the Crowdfundplugin.cs file, depending on the settings.
In this file, we don't have access to controller to get the Url.Action method.
Would need extra code more complicated I think...

Comment (dennisreimann) [reply BTCPayServer/wwwroot/crowdfund/app.js]: I updated it in d3dab94.

**Review by dennisreimann — CHANGES_REQUESTED** [review]
Currently it only works for the "no perks" case. If there are perks, they need to be selected first and afterwards the form would need to be filled.

**Comment by Nisaba** [comment]
OK, noted.
I will fix this before the end of this week.

**Comment by dennisreimann** [comment]
Please also add a test for both cases (pers and no perks), so that we can ensure the crowdfunding with form case will also be considered moving forward. Imho that's important, because the flow is very JS heavy and can easily be missed.

**Comment by Nisaba** [comment]
@dennisreimann fix for the perk has been made. You can check it.
I have to add tests; i hope this week-end...

**Review by dennisreimann — APPROVED** [review]
tACK. Thanks for the contribution!

**Review by dstrukt — APPROVED** [review]
tACK - LGTM.

+1 Thank you for the contribution - great work!

**Review by Kukks — APPROVED** [review]

**Pull request merged**